### ISSUE TITLE
rtdetr   export onnx  some error

### ISSUE THREAD

**Issue by QIANXUNZDL123** [ultralytics/ultralytics#3540]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

i use rtdetr-l.pt export to onnx  but have some  error.....    look here
from ultralytics import RTDETR
import torch
import numpy as np
import onnxruntime

def to_numpy(tensor):
    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()

model = RTDETR("rtdetr-l.pt")
model.export(format='onnx', dynamic=True)

print("ok1")

![image](https://github.com/ultralytics/ultralytics/assets/46549527/718502bb-707e-45b2-bf4b-52c1480ab9d0)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@QIANXUNZDL123 RTDETR is not compatible with `dynamic=True`, please export with default `dynamic=False`.

**Comment by glenn-jocher** [comment]
@QIANXUNZDL123 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. ONNX export works correctly for me, I'm unable to reproduce any issues:

<img width="915" alt="Screenshot 2023-07-05 at 13 34 30" src="https://github.com/ultralytics/ultralytics/assets/26833433/b2a315fa-046f-4082-b99a-65379cd37f5a">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by QIANXUNZDL123** [comment]
> @QIANXUNZDL123 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. ONNX export works correctly for me, I'm unable to reproduce any issues:
> 
> <img alt="Screenshot 2023-07-05 at 13 34 30" width="915" src="https://user-images.githubusercontent.com/26833433/251137375-b2a315fa-046f-4082-b99a-65379cd37f5a.png">
> We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.
> 
> ## How to create a Minimal, Reproducible Example
> When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:
> 
> * ✅ **Minimal** – Use as little code as possible to produce the problem
> * ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
> * ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem
> 
> For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:
> 
> * ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
> * ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.
> 
> If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem.
> 
> Thank you! 😃

opset must be set up 16, and the error has disappeared........................

**Comment by glenn-jocher** [comment]
Hi @QIANXUNZDL123! 👋 Thank you for providing the update on the ONNX export issue with YOLOv8. It's great to hear that setting the `opset` to 16 resolved the error for you. This information can be helpful to other users who might encounter a similar problem.

If you have any more questions or need further assistance, feel free to ask. We're here to help! 😃
